### PR TITLE
Unnecessary toast is diplaying

### PR DESCRIPTION
**Pull request by arpit01923**
Removed unnecessary toast without any action by user  -

**Issue**
Open #458 

**What is the change?**
- when a user goes to status page an unnecessary toast is shown because user does not have authorization and "/users/self" API will hit which requires authorization and give toastify error

**Is Development Tested?**
- No

**Before / After Change Screenshots**

Before-
image

After-
image

### CODE REVIEW

**Comment by shubhamsigdar1** [comment]
> Removed unnecessary toast without any action by user image

Can you please write description in this format: https://github.com/Real-Dev-Squad/website-www/pull/442#issue-1640507598

File: src/context/isUserAuthorized.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ const IsUserAuthorizedContext: FC<Props> = ({ children }) => {
         const { adminUser, superUser } = userRoles;
         setIsUserAuthorized(!!adminUser || !!superUser);
       } catch (err: any) {
-        toast(ERROR, err.message);
+        console.error(err)
```
Comment (shmbajaj) [inline src/context/isUserAuthorized.tsx:27]: Why error is ignored here?
Can't be checked for error mentioned in description and ignore only for that?

File: src/helperFunctions/getUser.ts [hunk 2]
```diff
@@ -11,7 +10,7 @@ const userData = async () => {
     const response = await requestPromise;
     return response.data;
   } catch (error) {
-    toast(ToastTypes.ERROR, (error as Error).message);
+    console.error(error)
```
Comment (shmbajaj) [inline src/helperFunctions/getUser.ts:13]: Check Above Comment.

**Review by shmbajaj — CHANGES_REQUESTED** [review]

File: src/context/isUserAuthorized.tsx (hunk 1 above)
Comment (arpit01923) [reply src/context/isUserAuthorized.tsx:27]: Issue is, for "/users/self" API, user must be authenticated. The toast is shown when user is not signin.

**Review by sahsisunny — COMMENTED** [review]
can we show custom error message ?

**Comment by arpit01923** [comment]
No, just remove the toastify issue and add error message to console.

File: src/context/isUserAuthorized.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ const IsUserAuthorizedContext: FC<Props> = ({ children }) => {
         const { adminUser, superUser } = userRoles;
         setIsUserAuthorized(!!adminUser || !!superUser);
       } catch (err: any) {
```
[the remark this answers is not in the document]
Comment (arpit01923) [reply src/context/isUserAuthorized.tsx:26]: sure

File: src/context/isUserAuthorized.tsx [hunk 4]
```diff
@@ -18,23 +18,23 @@ interface Props {
...
-                    adminUser: data.roles?.admin,
-                    superUser: data.roles?.super_user,
-                };
-                const { adminUser, superUser } = userRoles;
-                setIsUserAuthorized(!!adminUser || !!superUser);
-            } catch (err: any) {
-                toast(ERROR, err.message);
-            }
+  const [isUserAuthorized, setIsUserAuthorized] = useState<boolean>(false);
+  useEffect(() => {
```
Comment (shmbajaj) [inline src/context/isUserAuthorized.tsx:22]: What's your thought on replacing this with a redux toolkit?

File: src/context/isUserAuthorized.tsx [hunk 5]
```diff
@@ -18,23 +18,23 @@ interface Props {
...
+        const { data } = await requestPromise;
+        const userRoles = {
+          adminUser: data.roles?.admin,
+          superUser: data.roles?.super_user,
         };
-        fetchData();
+        const { adminUser, superUser } = userRoles;
+        setIsUserAuthorized(!!adminUser || !!superUser);
+      } catch (err) {
+        console.error(err);
```
Comment (shmbajaj) [inline src/context/isUserAuthorized.tsx:34]: Can you set the state `isUserAuth...` to false here?

File: src/components/footer/footer.module.scss [hunk 6]
```diff
@@ -1,9 +1,9 @@
-.infoRepo {
-    font-weight: 100;
-    margin-top: auto;
-    text-align: center;
-    padding: 15px 20px;
-    width: 100%;
-    position: relative;
-    bottom: 0;
-}
+.infoRepo {
```
Comment (shmbajaj) [inline src/components/footer/footer.module.scss:1]: Please revert this change, your editor styles are overriding repo styles.

File: src/helperFunctions/getUser.ts [hunk 7]
```diff
@@ -1,19 +1,18 @@
...
-        const response = await requestPromise;
-        return response.data;
-    } catch (error) {
-        toast(ToastTypes.ERROR, (error as Error).message);
-        return error;
-    }
+  try {
+    const { requestPromise } = fetch({
+      url: USER_SELF,
+      method: 'get',
```
Comment (shmbajaj) [inline src/helperFunctions/getUser.ts:8]: `get` is by default method, no need to explicit mention here.

File: .eslintrc.json [hunk 8]
```diff
@@ -17,7 +17,7 @@
     },
     "plugins": ["react", "@typescript-eslint"],
     "rules": {
-        "linebreak-style": ["error", "unix"],
+        "linebreak-style": ["error", "windows"],
```
Comment (shmbajaj) [inline .eslintrc.json:20]: Why change of rule here?

**Review by shmbajaj — CHANGES_REQUESTED** [review]

File: src/context/isUserAuthorized.tsx (hunk 4 above)
Comment (shmbajaj) [reply src/context/isUserAuthorized.tsx:22]: I guess it's better to use redux toolkit here.

**Review by shmbajaj — CHANGES_REQUESTED** [review]

File: src/helperFunctions/getUser.ts [hunk 9]
```diff
@@ -1,19 +1,17 @@
 import { USER_SELF } from '@/components/constants/url';
 import fetch from '@/helperFunctions/fetch';
-import { toast, ToastTypes } from '@/helperFunctions/toast';
 
 const userData = async () => {
```
Comment (shmbajaj) [inline src/helperFunctions/getUser.ts:4]: Can you add the return type here using typescript?
so, anyone using this function can get to know what to expect:
- If success
- If failure

**OR**

Create a separate PR, to use redux toolkit which replaces this function and the `useEffect`.

What's your thought @arpit01923 
cc: @Maheima

**Review by shmbajaj — APPROVED** [review]
LGTM ✅

File: src/helperFunctions/getUser.ts (hunk 9 above)
Comment (shmbajaj) [reply src/helperFunctions/getUser.ts:4]: Issue: https://github.com/Real-Dev-Squad/website-status/issues/492

**Review by Maheima — APPROVED** [review]
LGTM
